internal/resources/cluster.go

```diff
@@ -835,3 +863,54 @@ func initializePendingClusterWithPlanAndId(plan providerschema.Cluster, id strin
 	}
 	return plan
 }
+
+func (c *Cluster) checkDisk(plan providerschema.Cluster) error {
```

Comment:
This logic should be on V4 API not terraform.

Reply:
We had a long discussion on this and decided to not touch v4 API, and do the change on terraform. CC: @a-atri @rajagp @Talina06

Reply:
On second thought API server already does disk validation.  So blocking from client is valid approach.